rir/src/utils/Set.h

```diff
@@ -0,0 +1,65 @@
+#ifndef RIR_SET_H
```

Comment:
Why???

Reply:
* std::set :  needs "operator<" implemented and is generally big overhead
* std::unordered_set :  needs hash implemented, easy to get wrong, big overhead for small sets
* SmallSet, just needs `operator==` and is faster and less memory overhead for small sets

Reply:
just wanted something simpler...